### ISSUE TITLE
inferior-ess-prompt doesn't recognise prompt in for-loops

### ISSUE THREAD

**Issue by BrendanHalpin** [emacs-ess/ess-stata-mode#4]
inferior-ess-prompt doesn't work in for-loops, where the prompt takes this form:

```stata
forvalues x=1/2 {
  2. 
```
Changing ess-stata-mode.el from
```emacs-lisp
(inferior-ess-prompt           . "\\([.:>] \\|--more--\\)")
```
to 
```emacs-lisp
(inferior-ess-prompt           .  "\\( *[0-9]*[.:>] \\|--more--\\)")
```

resolves the problem (but I haven't tested much for unintended consequences).

**Comment by vspinu** [comment]
Thanks @BrendanHalpin for all the issues. Unfortunately AFAIK none of the ESS developers is using stata and can systematically develop for it. If you can provide PRs for all of your issues we would be more than happy to merge.

**Comment by BrendanHalpin** [comment]
I'll do that. 

Is it better to do one pull request per issue, or to pool them (the resolved ones at least)?

**Comment by lionel-** [comment]
One per issue please. PRs should be small and have only one concern.

**Comment by BrendanHalpin** [comment]
OK. Do I need some sort of write-access to the repository?

**Comment by lionel-** [comment]
You're good to go, anyone can submit a pull request. It will be reviewed and merged by someone with commit access.

**Comment by BrendanHalpin** [comment]
Closed by #7
